compiler/jsgen.nim

```diff
@@ -1623,22 +1623,56 @@ proc genToArray(p: PProc; n: PNode; r: var TCompRes) =
     localError(x.info, "'toArray' needs an array literal")
   r.res.add(")")
 
+proc genReprAux(p: PProc, n: PNode, r: var TCompRes, magic:string, typ:Rope = nil) =
```

Comment:
Please be consistent with your coding style. There is a lot of missing whitespace after the ``:`` in a lot of your code.